src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -329,11 +334,47 @@ const getCustomElementCustomTransformer = (
     style: 'static',
     styleImportData: 'queryparams',
   };
-  return [
+  const beforeCustomTransformers = [
     addDefineCustomElementFunctions(compilerCtx, components, outputTarget),
     updateStencilCoreImports(transformOpts.coreImportPath),
+  ];
+
+  if (config.transformAliasedImportPaths) {
+    beforeCustomTransformers.push(rewriteAliasedSourceFileImportPaths);
+  }
```

Comment:
checking my understanding: this is because the `updateStencilCoreImports` transformer has to run before the `rewriteAliasedSourceFileImportPaths` transformer in order to avoid the reported bug?

Reply:
Yup - exactly.

In the future it might be interesting to try to remove the need for `updateStencilCoreImports` and avoid subtle-ish bugs like this. I thought about pulling that thread, but felt that would have been much bigger in scope 😅